File: web/src/test/java/no/nav/sbl/dialogarena/rest/feil/ApplicationExceptionMapperTest.java
package no.nav.sbl.dialogarena.rest.feil;

import no.nav.sbl.dialogarena.sendsoknad.domain.exception.AuthorizationException;
import no.nav.sbl.dialogarena.sendsoknad.domain.exception.EttersendelseSendtForSentException;
import no.nav.sbl.dialogarena.sendsoknad.domain.exception.OpplastingException;
import no.nav.sbl.dialogarena.sendsoknad.domain.exception.SosialhjelpSoknadApiException;
import no.nav.sbl.dialogarena.sendsoknad.domain.exception.UgyldigOpplastingTypeException;
import no.nav.sbl.dialogarena.soknadinnsending.consumer.exceptions.SikkerhetsBegrensningException;
import no.nav.sbl.dialogarena.soknadinnsending.consumer.exceptions.TjenesteUtilgjengeligException;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.runners.MockitoJUnitRunner;

import javax.ws.rs.core.Response;

import static no.nav.sbl.dialogarena.rest.feil.Feilmelding.NO_BIGIP_5XX_REDIRECT;
import static org.assertj.core.api.Assertions.assertThat;

@RunWith(MockitoJUnitRunner.class)
public class ApplicationExceptionMapperTest {

    ApplicationExceptionMapper mapper = new ApplicationExceptionMapper();

    @Test
    public void skalGi415UnsupportedMediaTypeVedUgyldigOpplastingTypeException() {
        Response response = mapper.toResponse(new UgyldigOpplastingTypeException("feil", new RuntimeException(), "id"));
        assertThat(response.getStatus()).isEqualTo(415);
    }

    @Test
    public void skalGi413RequestTooLongVedOpplastingException() {
        Response response = mapper.toResponse(new OpplastingException("feil", new RuntimeException(), "id"));
        assertThat(response.getStatus()).isEqualTo(413);
    }

    @Test
    public void skalGi403ForbiddenVedAuthorizationException() {
        Response response = mapper.toResponse(new AuthorizationException("feil"));
        assertThat(response.getStatus()).isEqualTo(403);
    }

    @Test
    public void skalGi500MedHeaderForIngenBigIpRedirectForAndreKjenteUnntak() {
        Response response = mapper.toResponse(new SosialhjelpSoknadApiException("feil"));
        assertThat(response.getStatus()).isEqualTo(500);
        assertThat(response.getHeaderString(NO_BIGIP_5XX_REDIRECT)).isEqualTo("true");
    }

    @Test
    public void skalGi500MedHeaderForIngenBigIpRedirectVedTjenesteUtilgjengeligException() {
        Response response = mapper.toResponse(new TjenesteUtilgjengeligException("feil", new RuntimeException()));
        assertThat(response.getStatus()).isEqualTo(500);
        assertThat(response.getHeaderString(NO_BIGIP_5XX_REDIRECT)).isEqualTo("true");
    }

    @Test
    public void skalGi500MedHeaderForIngenBigIpRedirectVedEttersendelseSendtForSentException() {
        Response response = mapper.toResponse(new EttersendelseSendtForSentException("feil"));
        assertThat(response.getStatus()).isEqualTo(500);
        assertThat(response.getHeaderString(NO_BIGIP_5XX_REDIRECT)).isEqualTo("true");
    }

    @Test
    public void skalGi500VMedHeaderForIngenBigIpRedirectedSikkerhetsBegrensningException() {
        Response response = mapper.toResponse(new SikkerhetsBegrensningException("feil", new RuntimeException()));
        assertThat(response.getStatus()).isEqualTo(500);
        assertThat(response.getHeaderString(NO_BIGIP_5XX_REDIRECT)).isEqualTo("true");
    }
}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/app/feil/ApplicationExceptionMapperTest.kt
//package no.nav.sosialhjelp.soknad.app.feil
//
//import no.nav.sosialhjelp.soknad.app.exceptions.AuthorizationException
//import no.nav.sosialhjelp.soknad.app.exceptions.EttersendelseSendtForSentException
//import no.nav.sosialhjelp.soknad.app.exceptions.SikkerhetsBegrensningException
//import no.nav.sosialhjelp.soknad.app.exceptions.SosialhjelpSoknadApiException
//import no.nav.sosialhjelp.soknad.app.exceptions.TjenesteUtilgjengeligException
//import no.nav.sosialhjelp.soknad.app.rest.feil.ApplicationExceptionMapper
//import no.nav.sosialhjelp.soknad.app.exceptions.Feilmelding.Companion.NO_BIGIP_5XX_REDIRECT
//import no.nav.sosialhjelp.soknad.vedlegg.exceptions.OpplastingException
//import no.nav.sosialhjelp.soknad.vedlegg.exceptions.UgyldigOpplastingTypeException
//import org.assertj.core.api.Assertions.assertThat
//import org.junit.jupiter.api.Test
//
//internal class ApplicationExceptionMapperTest {
//
//    val mapper = ApplicationExceptionMapper()
//
//    @Test
//    fun skalGi415UnsupportedMediaTypeVedUgyldigOpplastingTypeException() {
//        val response = mapper.toResponse(UgyldigOpplastingTypeException("feil", RuntimeException(), "id"))
//        assertThat(response.status).isEqualTo(415)
//    }
//
//    @Test
//    fun skalGi413RequestTooLongVedOpplastingException() {
//        val response = mapper.toResponse(OpplastingException("feil", RuntimeException(), "id"))
//        assertThat(response.status).isEqualTo(413)
//    }
//
//    @Test
//    fun skalGi403ForbiddenVedAuthorizationException() {
//        val response = mapper.toResponse(AuthorizationException("feil"))
//        assertThat(response.status).isEqualTo(403)
//    }
//
//    @Test
//    fun skalGi500MedHeaderForIngenBigIpRedirectForAndreKjenteUnntak() {
//        val response = mapper.toResponse(SosialhjelpSoknadApiException("feil"))
//        assertThat(response.status).isEqualTo(500)
//        assertThat(response.getHeaderString(NO_BIGIP_5XX_REDIRECT)).isEqualTo("true")
//    }
//
//    @Test
//    fun skalGi500MedHeaderForIngenBigIpRedirectVedTjenesteUtilgjengeligException() {
//        val response = mapper.toResponse(TjenesteUtilgjengeligException("feil", RuntimeException()))
//        assertThat(response.status).isEqualTo(500)
//        assertThat(response.getHeaderString(NO_BIGIP_5XX_REDIRECT)).isEqualTo("true")
//    }
//
//    @Test
//    fun skalGi500MedHeaderForIngenBigIpRedirectVedEttersendelseSendtForSentException() {
//        val response = mapper.toResponse(EttersendelseSendtForSentException("feil"))
//        assertThat(response.status).isEqualTo(500)
//        assertThat(response.getHeaderString(NO_BIGIP_5XX_REDIRECT)).isEqualTo("true")
//    }
//
//    @Test
//    fun skalGi500VMedHeaderForIngenBigIpRedirectedSikkerhetsBegrensningException() {
//        val response = mapper.toResponse(SikkerhetsBegrensningException("feil", RuntimeException()))
//        assertThat(response.status).isEqualTo(500)
//        assertThat(response.getHeaderString(NO_BIGIP_5XX_REDIRECT)).isEqualTo("true")
//    }
//}
